src/cox.jl

```diff
@@ -224,6 +228,9 @@ Cox proportional hazard model estimate of coefficients. Returns a `CoxModel`
 object.
 """
 function StatsAPI.fit(::Type{CoxModel}, M::AbstractMatrix, y::AbstractVector; tol=1e-4, l2_cost=0)
+    if rank(M) < min(size(M)...)
+        throw(ArgumentError("model matrix is not full rank; some terms may be collinear"))
```

Comment:
Alternatively, we could add a `dropcollinear` keyword argument to mimic GLM. This is probably hot garbage but I think we could implement it naively within the existing framework as
```julia
r = rank(M)
if r < min(size(M)...)
    if dropcollinear
        F = qr(M, ColumnNorm())
        pivot = F.jpvt
        keepcols = pivot[1:r]
        dropcols = pivot[(r + 1):end]
        M = M[:, sort(keepcols)]
    else
        throw(ArgumentError("model matrix is not full rank and `dropcollinear=false`"))
    end
end
```
We could then use `dropcols` to either go back and re-add terms with a coefficient of 0 or emit some kind of informative warning.

Reply:
For reference, R's `coxph` fits the model and keeps around the redundant terms but the result has all `NA` for the coefficients and statistics and whatnot for those terms.